src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -84,26 +97,69 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
   } = useVenueEvents({
     venueIds: relatedVenueIds,
   });
-
   const isLoadingSchedule = isLoading || isEventsLoading;
 
   const [selectedDayIndex, setSelectedDayIndex] = useState(0);
 
+  const liveAndFutureEvents = useMemo(
+    () =>
+      relatedVenueEvents.map((event) =>
+        secondsToMilliseconds(
+          event.start_utc_seconds + minutesToSeconds(event.duration_minutes)
+        ) > startOfToday().getTime()
+          ? event.start_utc_seconds
+          : Number.MAX_SAFE_INTEGER
+      ),
+    [relatedVenueEvents]
+  );
+
+  const minDate = useMemo(
+    () =>
+      liveAndFutureEvents.length
+        ? Math.min(...liveAndFutureEvents)
+        : minRangeValue,
+    [liveAndFutureEvents]
+  );
+
+  const maxDate = useMemo(
+    () =>
+      Math.max(
+        ...relatedVenueEvents.map(
+          (event) =>
+            event.start_utc_seconds + minutesToSeconds(event.duration_minutes)
+        ),
+        maxRangeValue
+      ),
+    [relatedVenueEvents]
+  );
+
+  const dayDifference = differenceInDays(
+    fromUnixTime(maxDate),
+    fromUnixTime(minDate)
+  );
+
   const weekdays = useMemo(() => {
     const formatDayLabel = (day: Date | number) => {
       if (isScheduleTimeshifted) {
-        return format(day, "E, LLL d");
+        return format(day, "do");
       } else {
         return formatDateRelativeToNow(day, {
-          formatOtherDate: (dateOrTimestamp) => format(dateOrTimestamp, "E"),
+          formatOtherDate: (dateOrTimestamp) => format(dateOrTimestamp, "do"),
+          formatTomorrow: (dateOrTimestamp) => format(dateOrTimestamp, "do"),
         });
       }
     };
 
-    return range(SCHEDULE_SHOW_DAYS_AHEAD).map((dayIndex) => {
-      const day = addDays(firstDayOfSchedule, dayIndex);
+    return range(dayDifference).map((dayIndex) => {
+      const day = addDays(todaysDate, dayIndex);
```

Comment:
i don't think it's correct.
let's say we have events from 1 to 2 of September, dayDifference = 2
but instead we'll show only today, tomorrow (which is not correct - these dates do not have events).

Reply:
fixed in 85038ed